### PR TITLE
shell: respect WAYLAND_DISPLAY

### PR DESCRIPTION
**Pull request by Saviq**
For automated test purposes, we need to be able to tell, which WAYLAND_DISPLAY to use.

### CODE REVIEW

File: glue/bin/run-shell [hunk 1]
```diff
@@ -29,6 +29,11 @@ export XDG_RUNTIME_DIR="$(dirname $XDG_RUNTIME_DIR)"
 mkdir -p "$XDG_RUNTIME_DIR" -m 700
 mkdir -p "$XDG_CACHE_HOME"
 
+if [ -n "${WAYLAND_DISPLAY}" ] && [ -e "${XDG_RUNTIME_DIR}/${WAYLAND_DISPLAY}" ] || [ -e "${XDG_RUNTIME_DIR}/${WAYLAND_DISPLAY}.lock" ]; then
+    echo "ERROR: ${XDG_RUNTIME_DIR}/${WAYLAND_DISPLAY} is busy, please choose a different XDG_RUNTIME_DIR or WAYLAND_DISPLAY"
+    exit 1
+fi
+
```
Comment (AlanGriffiths) [inline glue/bin/run-shell]: I don't think this is needed or wanted.

If there's an existing `WAYLAND_DISPLAY` served by another compositor Miriway will use that as host. There's no need to complain about it.
```suggestion
```

File: glue/bin/run-shell [hunk 2]
```diff
@@ -158,6 +163,6 @@ EOT
 done
 
 cd "$HOME"
-unset WAYLAND_DISPLAY
```
Comment (AlanGriffiths) [inline glue/bin/run-shell:161]: Yes. Not sure why that was there

**Review by AlanGriffiths — CHANGES_REQUESTED** [review]
I think this does more than intended

File: glue/bin/run-shell (hunk 1 above)
Comment (Saviq) [reply glue/bin/run-shell]: Ah!

**Review by AlanGriffiths — APPROVED** [review]
bors merge

**Pull request merged**